From dfb4de0b7e957f77ff389ffa8243a3ce1bb7184d Mon Sep 17 00:00:00 2001
From: AD-Aleksandrov <aleksandr.aleksandrov@emlid.com>
Date: Fri, 24 Nov 2017 20:13:02 +0300
Subject: [PATCH] arm: dts: allwinner: update DT for emlid-iota, new overlays

- add spidev0, spidev1 to emlid-iota DT file
- new overlays to enable spi for sun50i-h5, sun8i-h3
---
 arch/arm/boot/dts/overlay/Makefile                 |  2 +
 .../boot/dts/overlay/sun8i-h3-spi0-status-okay.dts | 13 +++++++
 .../boot/dts/overlay/sun8i-h3-spi1-status-okay.dts | 13 +++++++
 arch/arm64/boot/dts/allwinner/overlay/Makefile     |  2 +
 .../overlay/sun50i-h5-spi0-status-okay.dts         | 13 +++++++
 .../overlay/sun50i-h5-spi1-status-okay.dts         | 13 +++++++
 .../boot/dts/allwinner/sun50i-h5-emlid-iota.dts    | 44 ++++++++++++++++++++++
 drivers/spi/spidev.c                               |  1 +
 8 files changed, 101 insertions(+)
 create mode 100644 arch/arm/boot/dts/overlay/sun8i-h3-spi0-status-okay.dts
 create mode 100644 arch/arm/boot/dts/overlay/sun8i-h3-spi1-status-okay.dts
 create mode 100644 arch/arm64/boot/dts/allwinner/overlay/sun50i-h5-spi0-status-okay.dts
 create mode 100644 arch/arm64/boot/dts/allwinner/overlay/sun50i-h5-spi1-status-okay.dts

diff --git a/arch/arm/boot/dts/overlay/Makefile b/arch/arm/boot/dts/overlay/Makefile
index 2bcbcee..e06a0d6 100644
--- a/arch/arm/boot/dts/overlay/Makefile
+++ b/arch/arm/boot/dts/overlay/Makefile
@@ -60,6 +60,8 @@ dtbo-$(CONFIG_MACH_SUN8I) += \
 	sun8i-h3-spi-add-cs1.dtbo \
 	sun8i-h3-spi-jedec-nor.dtbo \
 	sun8i-h3-spi-spidev.dtbo \
+	sun8i-h3-spi0-status-okay.dtbo \
+	sun8i-h3-spi1-status-okay.dtbo \
 	sun8i-h3-uart1.dtbo \
 	sun8i-h3-uart2.dtbo \
 	sun8i-h3-uart3.dtbo \
diff --git a/arch/arm/boot/dts/overlay/sun8i-h3-spi0-status-okay.dts b/arch/arm/boot/dts/overlay/sun8i-h3-spi0-status-okay.dts
new file mode 100644
index 0000000..177381a
--- /dev/null
+++ b/arch/arm/boot/dts/overlay/sun8i-h3-spi0-status-okay.dts
@@ -0,0 +1,13 @@
+/dts-v1/;
+/plugin/;
+
+/ {
+    compatible = "allwinner,sun8i-h3";
+
+    fragment@0 {
+        target = <&spi0>;
+        __overlay__ {
+            status = "okay";
+        };
+    };
+};
diff --git a/arch/arm/boot/dts/overlay/sun8i-h3-spi1-status-okay.dts b/arch/arm/boot/dts/overlay/sun8i-h3-spi1-status-okay.dts
new file mode 100644
index 0000000..1e7d639
--- /dev/null
+++ b/arch/arm/boot/dts/overlay/sun8i-h3-spi1-status-okay.dts
@@ -0,0 +1,13 @@
+/dts-v1/;
+/plugin/;
+
+/ {
+    compatible = "allwinner,sun8i-h3";
+
+    fragment@0 {
+        target = <&spi1>;
+        __overlay__ {
+            status = "okay";
+        };
+    };
+};
diff --git a/arch/arm64/boot/dts/allwinner/overlay/Makefile b/arch/arm64/boot/dts/allwinner/overlay/Makefile
index 379d852..511eee8 100644
--- a/arch/arm64/boot/dts/allwinner/overlay/Makefile
+++ b/arch/arm64/boot/dts/allwinner/overlay/Makefile
@@ -21,6 +21,8 @@ dtbo-$(CONFIG_ARCH_SUNXI) += \
 	sun50i-h5-spi-add-cs1.dtbo \
 	sun50i-h5-spi-jedec-nor.dtbo \
 	sun50i-h5-spi-spidev.dtbo \
+	sun50i-h5-spi0-status-okay.dtbo \
+	sun50i-h5-spi1-status-okay.dtbo \
 	sun50i-h5-uart1.dtbo \
 	sun50i-h5-uart2.dtbo \
 	sun50i-h5-uart3.dtbo \
diff --git a/arch/arm64/boot/dts/allwinner/overlay/sun50i-h5-spi0-status-okay.dts b/arch/arm64/boot/dts/allwinner/overlay/sun50i-h5-spi0-status-okay.dts
new file mode 100644
index 0000000..df88928
--- /dev/null
+++ b/arch/arm64/boot/dts/allwinner/overlay/sun50i-h5-spi0-status-okay.dts
@@ -0,0 +1,13 @@
+/dts-v1/;
+/plugin/;
+
+/ {
+    compatible = "allwinner,sun50i-h5";
+
+    fragment@0 {
+        target = <&spi0>;
+        __overlay__ {
+            status = "okay";
+        };
+    };
+};
diff --git a/arch/arm64/boot/dts/allwinner/overlay/sun50i-h5-spi1-status-okay.dts b/arch/arm64/boot/dts/allwinner/overlay/sun50i-h5-spi1-status-okay.dts
new file mode 100644
index 0000000..ba93b88
--- /dev/null
+++ b/arch/arm64/boot/dts/allwinner/overlay/sun50i-h5-spi1-status-okay.dts
@@ -0,0 +1,13 @@
+/dts-v1/;
+/plugin/;
+
+/ {
+    compatible = "allwinner,sun50i-h5";
+
+    fragment@0 {
+        target = <&spi1>;
+        __overlay__ {
+            status = "okay";
+        };
+    };
+};
diff --git a/arch/arm64/boot/dts/allwinner/sun50i-h5-emlid-iota.dts b/arch/arm64/boot/dts/allwinner/sun50i-h5-emlid-iota.dts
index 014f772..a6a97175 100644
--- a/arch/arm64/boot/dts/allwinner/sun50i-h5-emlid-iota.dts
+++ b/arch/arm64/boot/dts/allwinner/sun50i-h5-emlid-iota.dts
@@ -146,3 +146,47 @@
 	/* usb0_vbus-supply = <&reg_usb0_vbus>; */
 	status = "okay";
 };
+
+&pio {
+    spi0_cs_pins: spi0_cs_pins {
+        pins = "PC3";
+        function = "gpio_out";
+    };
+
+    spi1_cs_pins: spi1_cs_pins {
+        pins = "PA13";
+        function = "gpio_out";
+    };
+};
+
+&spi0 {
+    #address-cells = <1>;
+    #size-cells = <0>;
+
+    status = "disabled";
+    pinctrl-names = "default";
+    pinctrl-0 = <&spi0_pins &spi0_cs_pins>;
+    cs-gpios = <&pio 2 3 GPIO_ACTIVE_HIGH>;    /* PC3 */
+
+    spidev0: spi@0 {
+        compatible = "spidev";
+        reg = <0>;
+        spi-max-frequency = <25000000>;
+    };
+};
+
+&spi1 {
+    #address-cells = <1>;
+    #size-cells = <0>;
+
+    status = "disabled";
+    pinctrl-names = "default";
+    pinctrl-0 = <&spi1_pins &spi1_cs_pins>;
+    cs-gpios = <&pio 0 13 GPIO_ACTIVE_HIGH>;    /* PA13 */
+
+    spidev1: spi@1 {
+        compatible = "spidev";
+        reg = <0>;
+        spi-max-frequency = <25000000>;
+    };
+};
diff --git a/drivers/spi/spidev.c b/drivers/spi/spidev.c
index cda1071..4f3779d 100644
--- a/drivers/spi/spidev.c
+++ b/drivers/spi/spidev.c
@@ -669,6 +669,7 @@ static const struct of_device_id spidev_dt_ids[] = {
 	{ .compatible = "lineartechnology,ltc2488" },
 	{ .compatible = "ge,achc" },
 	{ .compatible = "semtech,sx1301" },
+	{ .compatible = "spidev" },
 	{},
 };
 MODULE_DEVICE_TABLE(of, spidev_dt_ids);
-- 
2.7.4

